### DISCUSSION TITLE
Issue search lacks support for logical-OR including multiple milestones

### DISCUSSION THREAD

**Discussion by ericstj** [community/community#63167] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

Issue search allows for searching for labels using OR syntax, now that lists for `label:a,b,c` and `-label:x,y,z` added.  This is GREAT!

What's lacking is the same support for `milestone`.  I'd like to be able to do `milestone:8.0.0,8.0.1,none` where `none` is some marker for `no:milestone`.  Today search only allows queries for a single milestone or none at all.  Support for lists here can help make a single query for the set of issues - which might be interesting during a product release phase or a bug triaging cleanup effort.

Thanks!

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by SAMIULLAH-KALHORO-DEV** [comment]
Subject: Request for Enhanced Milestone Search Functionality

Dear @ericstj 

I hope this message finds you well. I wanted to provide some feedback regarding the recent improvements in issue search functionality on the platform. The addition of the OR syntax for label searches, particularly the use of lists such as `label:a,b,c` and `-label:x,y,z`, has been a fantastic enhancement. This enhancement has significantly improved our ability to manage and organize issues effectively.

However, I would like to bring up a point regarding the search functionality for milestones. Currently, the search system only allows for queries involving a single milestone or none at all. While this has been functional for our purposes so far, I believe there's an opportunity to enhance the search capabilities for milestones.

Specifically, I would like to suggest the addition of support for lists when searching for milestones. This would allow users to query for a range of milestones or even specify a marker for issues that have no milestone assigned. For example, being able to search using a query like `milestone:8.0.0,8.0.1,none` would provide significant value, especially during product release phases or bug triaging cleanup efforts.

By implementing this enhancement, users could perform a single comprehensive search to gather a set of issues associated with multiple milestones and those without any milestone assigned. This would undoubtedly streamline our workflows and contribute to more efficient issue management.

I understand that implementing such features requires careful consideration and technical resources. However, I believe this enhancement aligns well with the current trajectory of improving search capabilities, which in turn enhances overall user experience and productivity. I wanted to take a moment to share this suggestion, as I believe it could greatly benefit our team and others who rely on the platform for issue tracking and management.

Thank you for your time and consideration. I look forward to any updates or insights you might have on this matter. If there's an appropriate channel or forum for further discussion, please do let me know.

Best regards,
